<h3 style="background-color:DodgerBlue;color:white;padding:20px;text-align:center;">
    HOMEWORK ASSIGNMENT <br>
    <h4><i>Class:</i> IMPH 4 </h4>
    <h4><i>Name:</i>  LE THI NGOC ANH</h4>
</h3>

In [1]:
local datapath C:\Users\DELL\Dropbox\Cigaret.csv
import delimited using "`datapath'", clear
set linesize 143
qui {
    la var state  "state"
    la var year   "year"
    la var cpi    "consummer price index"
    la var pop    "state population"
    la var packpc "number of packs per capita"
    la var income "state personal income (total, nominal)"
    la var tax    "average state, federal, and average local excise taxes for fiscal year"
    la var avgprs "average price during fiscal year, including sales taxes"
    la var taxs   "average excise taxes for fiscal year, including sales taxes"
}
encode state, gen(id)
describe
list in 1/5



(10 vars, 528 obs)



. encode state, gen(id)

. describe

Contains data
  obs:           528                          
 vars:            11                          
 size:        20,064                          
-----------------------------------------------------------------------------------------------------------------------------------------------
              storage   display    value
variable name   type    format     label      variable label
-----------------------------------------------------------------------------------------------------------------------------------------------
v1              int     %8.0g                 
state           str2    %9s                   state
year            int     %8.0g                 year
cpi             float   %9.0g                 consummer price index
pop             long    %12.0g                state population
packpc          float   %9.0g                 number of packs per capita
income          long    %12.0g          

In [2]:
global ylist packpc
global xlist tax avgprs taxs

<h4>Question: Analysis of drug consumption is with related factors?</h4>

In [3]:
// This dataset is the panel data
// 11 years from 1985 to 1995
// 48 states reported each year

// Declare panel data in Stata and summary data
xtset id year
xtdes
xtsum $ylist $xlist


       panel variable:  id (strongly balanced)
        time variable:  year, 1985 to 1995
                delta:  1 unit


      id:  1, 2, ..., 48                                     n =         48
    year:  1985, 1986, ..., 1995                             T =         11
           Delta(year) = 1 unit
           Span(year)  = 11 periods
           (id*year uniquely identifies each observation)

Distribution of T_i:   min      5%     25%       50%       75%     95%     max
                        11      11      11        11        11      11      11

     Freq.  Percent    Cum. |  Pattern
 ---------------------------+-------------
       48    100.00  100.00 |  11111111111
 ---------------------------+-------------
       48    100.00         |  XXXXXXXXXXX


Variable         |      Mean   Std. Dev.       Min        Max |    Observations
-----------------+--------------------------------------------+----------------
packpc   overall |   106.455   23.12423    49.2722    197.994 |  

In [4]:
// Differents model to estimate relation between drug consumption with associate factors
qui eststo model1: reg $ylist $xlist //Pooled regression model

qui eststo model2: xtreg $ylist $xlist, be //Between effect
qui eststo model3: xtreg $ylist $xlist, fe //Fixed effect
qui eststo model4: xtreg $ylist $xlist, re //Random effect

esttab model1 model2 model3 model4, ci r2 noconstant //Showing model's result
reg D.($ylist $xlist), noconstant //First differents
hausman model3 model4







--------------------------------------------------------------------------------------------------------------------
                                (1)                       (2)                       (3)                       (4)   
                             packpc                    packpc                    packpc                    packpc   
--------------------------------------------------------------------------------------------------------------------
tax                         -0.0683                    -0.199                    -0.267                    -0.259   
                     [-0.608,0.472]            [-2.193,1.796]           [-0.570,0.0356]           [-0.558,0.0401]   

avgprs                       -0.218***                 -1.132                    -0.270***                 -0.268***
                    [-0.309,-0.126]            [-2.394,0.129]           [-0.297,-0.242]           [-0.296,-0.241]   

taxs                         -0.446                     

In [5]:
xtreg $ylist $xlist, re theta
xttest0



Random-effects GLS regression                   Number of obs     =        528
Group variable: id                              Number of groups  =         48

R-sq:                                           Obs per group:
     within  = 0.8433                                         min =         11
     between = 0.2799                                         avg =       11.0
     overall = 0.3631                                         max =         11

                                                Wald chi2(3)      =    2570.89
corr(u_i, X)   = 0 (assumed)                    Prob > chi2       =     0.0000
theta          = .92498438

------------------------------------------------------------------------------
      packpc |      Coef.   Std. Err.      z    P>|z|     [95% Conf. Interval]
-------------+----------------------------------------------------------------
         tax |   -.258913   .1525743    -1.70   0.090    -.5579532    .0401272
      avgprs |  -.2684545   .0140034

<h3 style="background-color:DodgerBlue;color:white;padding:20px;line-height:1.5;">
Conclusion: <br>

<p style="background-color:DodgerBlue;color:white;padding:20px;line-height:1.5;">
Có sự khác biệt có ý nghĩa thống kê trong giá trung bình 1 bao thuốc với sự tiêu thụ thuốc lá. <br>
Mô hình regression cộng dồn ở 48 bang trong 11 năm cho thấy giá trung bình 1 bao thuốc tăng lên 1 đơn vị thì sự tiêu thụ giảm 22%.
Sự khác biệt từng năm: Qua từng năm, nếu tăng giá 1 bao thuốc trung bình (bao gồm thuế) lên 1 đơn vị thì sự tiêu thụ thuốc lá giảm 17% ở 48 bang. 
Giá trị p của kiểm định Hausman không cho thấy có ý nghĩa thống kê, như vậy lựa chọn mô hình fixed effect model. <br>
Mô hình fixed effect model cho thấy tăng 1 đơn vị giá bao thuốc thì sẽ giảm 27% lượng tiêu thụ thuốc lá. Mô hình giải thích 84.3% sự tiêu thụ thuốc lá ở các bang.
    </p>
</h3>